data-tool/flows/tombstone/tombstone_utils.py

```diff
@@ -157,40 +162,40 @@ def format_parties_data(data: dict) -> list[dict]:
 
         # Format party role
         formatted_party_roles = party['party_roles']
-        for _, r in group.iterrows():
 
-            if (role_code := r['cp_party_typ_cd']) not in role_mapping.keys():
-                continue
+        if (role_code := party_info['cp_party_typ_cd']) not in role_mapping.keys():
+            continue
 
-            role = role_mapping[role_code]  # Will raise KeyError if role_code not in mapping
+        role = role_mapping[role_code]  # Will raise KeyError if role_code not in mapping
 
-            party_role = copy.deepcopy(PARTY_ROLE)
-            party_role['role'] = role
+        party_role = copy.deepcopy(PARTY_ROLE)
+        party_role['role'] = role
 
-            appointment_date = None
-            if appointment_date := r['cp_appointment_dt_str']:
-                appointment_date = appointment_date + ' 00:00:00+00:00'
-            party_role['appointment_date'] = appointment_date
+        appointment_date = None
+        earliest_party_info = find_earliest_party(party_info, parties_data)
+        if appointment_date := earliest_party_info['cp_appointment_dt_str']:
```

Comment:
Does colin output show the appointment date from the earliest party?

Reply:
No, I don't find any in COLIN outputs. But appointment date affects the directors in outstanding ARs in LEAR (ref: https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/models/party_role.py#L141)

Reply:
And some latest parties don't have appointment date in place and we have to search backward to get the earliest results. For examples,
![image](https://github.com/user-attachments/assets/25250e17-947c-4e48-af96-1e90c7ad38bf)

Reply:
I thought it specific to `OFF` party

Reply:
Actually it excludes `OFF` party (compatible with `OFF` but not 100% accurate as only the latest `OFF` is selected in query)